movieDetails.js

```diff
@@ -0,0 +1,242 @@
+
+    ratingButton.addEventListener("click", () => {
+      movie.rating = i + 1;
+      updateStars(movieRatings, movie.rating);
+    });
+  }
+
+  movieExtraInfo.append(moviePrice, movieRatings, sendMessage);
+  movieCard.append(movieTitle, movieCardLayout, movieDirector, movieExtraInfo);
+
+  sendMessage.addEventListener("click", function (e) {
+    e.stopPropagation();
+    openCommentsPopup(movie);
+  });
+
+  movieCardLayout.addEventListener("click", function () {
+    openMovieDetailsPopup(movie);
+  });
+
+  updateStars(movieRatings, movie.rating || 0);
+
+  return movieCard;
+}
+
+// Update star rating
+function updateStars(ratingContainer, rating) {
+  const stars = ratingContainer.querySelectorAll(".rating i");
+  stars.forEach((star, index) => {
+    if (index < rating) {
+      star.classList.add("fa-solid");
+      star.classList.remove("fa-regular");
+    } else {
+      star.classList.add("fa-regular");
+      star.classList.remove("fa-solid");
+    }
+  });
+}
+
+// Popup and comment functionality 
+function openCommentsPopup(movie) {
+  const popupBackground = document.createElement("div");
+  popupBackground.classList.add("popup-background");
+
+  const popupContainer = document.createElement("div");
+  popupContainer.classList.add("popup-container");
+
+  const popupHeader = document.createElement("h2");
+  popupHeader.textContent = `Comments for ${movie.title}`;
+  popupHeader.classList.add("popup-header");
+
+  const closeIcon = document.createElement("i");
+  closeIcon.classList.add("fas", "fa-times", "close-icon");
+
+  const miniPoster = document.createElement("img");
+  miniPoster.src = movie.poster_url;
+  miniPoster.alt = `Poster for ${movie.title}`;
+  miniPoster.classList.add("mini-poster");
+
+  const commentsSection = document.createElement("div");
+  commentsSection.classList.add("comments-section");
+
+  if (movie.comments && movie.comments.length > 0) {
+    movie.comments.forEach(comment => {
+      const commentElement = document.createElement("p");
+      commentElement.textContent = comment;
+      commentsSection.appendChild(commentElement);
+    });
+  }
+
+  const commentInput = document.createElement("textarea");
+  commentInput.classList.add("comment-input");
+  commentInput.placeholder = "Write your comment here...";
+
+  const submitButton = document.createElement("button");
+  submitButton.textContent = "Submit";
+  submitButton.classList.add("submit-button");
+
+  const headerContainer = document.createElement("div");
+  headerContainer.classList.add("header-container");
+  headerContainer.append(miniPoster, popupHeader, closeIcon);
+
+  popupContainer.append(headerContainer, commentsSection, commentInput, submitButton);
+  popupBackground.appendChild(popupContainer);
+  document.body.appendChild(popupBackground);
+
+  closeIcon.addEventListener("click", function () {
+    document.body.removeChild(popupBackground);
+  });
+
+  submitButton.addEventListener("click", function () {
```

Comment:
This could be extracted in another function to not make `openCommentsPopup` so long.